crates/router/src/connector/braintree/braintree_graphql_transformers.rs

```diff
@@ -0,0 +1,988 @@
+    pub status: BraintreeRefundStatus,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct BraintreeRefundTransaction {
+    pub refund: Option<BraintreeRefundTransactionBody>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BraintreeRefundResponseData {
+    pub refund_transaction: Option<BraintreeRefundTransaction>,
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct BraintreeRefundResponse {
+    pub data: Option<BraintreeRefundResponseData>,
+    pub errors: Option<Vec<ErrorDetails>>,
+}
+
+impl TryFrom<types::RefundsResponseRouterData<api::Execute, BraintreeRefundResponse>>
+    for types::RefundsRouterData<api::Execute>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::RefundsResponseRouterData<api::Execute, BraintreeRefundResponse>,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            response: if item.response.errors.is_some() {
+                build_error_response(
+                    &item
+                        .response
+                        .errors
+                        .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?,
+                    item.http_code,
+                )
+            } else {
+                let refund_data = match &item.response.data {
+                    Some(refund_info) => refund_info
+                        .refund_transaction
+                        .as_ref()
+                        .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                        .refund
+                        .as_ref(),
+                    None => Err(errors::ConnectorError::ResponseDeserializationFailed)?,
+                };
+
+                Ok(types::RefundsResponseData {
+                    connector_refund_id: refund_data
+                        .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                        .id
+                        .clone(),
+                    refund_status: enums::RefundStatus::from(
+                        refund_data
+                            .as_ref()
+                            .ok_or(errors::ConnectorError::ResponseDeserializationFailed)?
+                            .status
+                            .clone(),
+                    ),
+                })
+            },
+            ..item.data
+        })
+    }
+}
+
+#[derive(Debug, Serialize)]
+pub struct BraintreeRSyncRequest {
+    query: String,
+}
+
+impl TryFrom<&types::RefundSyncRouterData> for BraintreeRSyncRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::RefundSyncRouterData) -> Result<Self, Self::Error> {
+        let refund_id = item.request.get_connector_refund_id()?;
+        let query = format!("query {{ search {{ refunds(input: {{ id: {{is: \"{}\"}} }}, first: 1) {{ edges {{ node {{ id status createdAt amount {{ value currencyCode }} orderId }} }} }} }} }}",refund_id);
```

Comment:
assign this to constant

Reply:
It cannot be assigned to a constant as the query is dynamic, we have to insert `refund_id` in the query